plugins/chat/app/services/chat/lookup_channel_threads.rb

```diff
@@ -93,28 +92,26 @@ def fetch_threads(guardian:, channel:, **)
             user: :user_status,
           ],
         )
-        .joins(:user_chat_thread_memberships, :original_message)
         .joins(
-          "LEFT JOIN chat_messages AS last_message ON last_message.id = chat_threads.last_message_id",
+          "LEFT JOIN user_chat_thread_memberships ON chat_threads.id = user_chat_thread_memberships.thread_id AND user_chat_thread_memberships.user_id = #{guardian.user.id} AND user_chat_thread_memberships.notification_level IN (#{::Chat::UserChatThreadMembership.notification_levels[:muted]})",
         )
-        .where("user_chat_thread_memberships.user_id = ?", guardian.user.id)
-        .where(
-          "user_chat_thread_memberships.notification_level IN (?)",
-          [
-            ::Chat::UserChatThreadMembership.notification_levels[:normal],
-            ::Chat::UserChatThreadMembership.notification_levels[:tracking],
-          ],
+        .joins(
+          "LEFT JOIN chat_messages AS last_message ON chat_threads.last_message_id = last_message.id",
+        )
+        .joins(
+          "INNER JOIN chat_messages AS original_message ON chat_threads.original_message_id = original_message.id",
         )
-        .where("chat_threads.channel_id = ?", channel.id)
+        .where(channel_id: channel.id)
+        .where("original_message.chat_channel_id = chat_threads.channel_id")
+        .where("original_message.deleted_at IS NULL")
+        .where("last_message.chat_channel_id = chat_threads.channel_id")
         .where("last_message.deleted_at IS NULL")
-        .limit(context.limit)
-        .offset(context.offset)
+        .where("chat_threads.replies_count > 0")
```

Comment:
It's a guard, a thread with no replies is not interesting to us.

Also note that I didn't create multiple index which could be useful here, will decide later to add them or not.